cloudinit/net/networkd.py

```diff
@@ -0,0 +1,229 @@
+
+    def __init__(self, config=None):
+        if not config:
+            config = {}
+        self.resolved_conf = config.get('resolved_conf_fn',
+                                        '/etc/systemd/resolved.conf')
+        self.network_conf_dir = config.get('network_conf_dir',
+                                           '/etc/systemd/network/')
+
+    def generate_match_section(self, iface, cfg):
+        sec = 'Match'
+        match_dict = {
+            'name': 'Name',
+            'driver': 'Driver',
+            'mac_address': 'MACAddress'
+        }
+
+        if not iface:
+            return
+
+        for k, v in match_dict.items():
+            if k in iface and iface[k]:
+                cfg.update_section(sec, v, iface[k])
+
+        return iface['name']
+
+    def generate_link_section(self, iface, cfg):
+        sec = 'Link'
+
+        if not iface:
+            return
+
+        if 'mtu' in iface and iface['mtu']:
+            cfg.update_section(sec, 'MTUBytes', iface['mtu'])
+
+    def parse_routes(self, conf, cfg):
+        sec = 'Route'
+        for k, v in conf.items():
+            if k == 'gateway':
+                cfg.update_section(sec, 'Gateway', v)
+            elif k == 'network':
+                tmp = v
+                if 'prefix' in conf:
+                    tmp += '/' + str(conf['prefix'])
+                cfg.update_section(sec, 'Destination', tmp)
+            elif k == 'metric':
+                cfg.update_section(sec, 'Metric', v)
+
+    def parse_subnets(self, iface, cfg):
+        dhcp = 'no'
+        for e in iface.get('subnets', []):
+            t = e['type']
+            if t == 'dhcp4' or t == 'dhcp':
+                if dhcp == 'no':
+                    dhcp = 'ipv4'
+                elif dhcp == 'ipv6':
+                    dhcp = 'yes'
+            elif t == 'dhcp6':
+                if dhcp == 'no':
+                    dhcp = 'ipv6'
+                elif dhcp == 'ipv4':
+                    dhcp = 'yes'
+            if 'routes' in e and e['routes']:
+                for i in e['routes']:
+                    self.parse_routes(i, cfg)
+            elif 'address' in e:
+                for k, v in e.items():
+                    if k == 'address':
+                        tmp = v
+                        if 'prefix' in e:
+                            tmp += '/' + str(e['prefix'])
+                        cfg.update_section('Address', 'Address', tmp)
+                    elif k == 'gateway':
+                        cfg.update_section('Route', 'Gateway', v)
+                    elif k == 'dns_nameservers':
+                        cfg.update_section('Network', 'DNS', ' '.join(v))
+                    elif k == 'dns_search':
+                        cfg.update_section('Network', 'Domains', ' '.join(v))
+
+        cfg.update_section('Network', 'DHCP', dhcp)
+
+    def dhcp_domain(self, d, cfg):
+        for item in ['dhcp4domain', 'dhcp6domain']:
```

Comment:
Yes, Photon has a newer version of systemd which supports [DHCPv4] & [DHCPv6]
For that I'm using this. If it's okay, let it be. We have some users who use this feature although not officially supported by network config netplan config was giving this info and we used it. It will be a lot of trouble explaining people why we can't support it anymore.

Reply:
So you're saying you're using our v2/netplan config, but you've added extra `dhcp4domain` and `dhcp6domain` keys to the config? Where did these keys come from? How is it different from the `dhcp4` and `dhcp6` keys?

It's not ideal. Can you add a comment to the method saying something like "This is to accommodate extra keys present in VMWare config"?

Reply:
Done